src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -22,7 +22,7 @@ protected internal JsonConverter()
             // In the future, this will be check for !IsSealed (and excluding value types).
             CanBePolymorphic = TypeToConvert == typeof(object);
             IsValueType = TypeToConvert.IsValueType;
-            HandleNullValue = ShouldHandleNullValue;
+            HandleNull = IsValueType;
```

Comment:
@steveharter What was the reason why we were caching `ShouldHandleNullValue` in `HandleNullValue`? Should we get rid of that?

Reply:
I wondered the same thing. I couldn't tell why we needed `ShouldHandleNullValue`.